stackslib/src/core/mempool.rs

```diff
@@ -1681,48 +1738,121 @@ impl MemPoolDB {
-                                fee_iterator.next().map_err(|err| Error::SqliteError(err))?
-                            } else {
+                    // When the retry list is empty, read from the mempool db depending on the configured miner strategy
+                    match settings.strategy {
+                        MemPoolWalkStrategy::GlobalFeeRate => {
+                            let start_with_no_estimate =
+                                tx_consideration_sampler.sample(&mut rng) < settings.consider_no_estimate_tx_prob;
+                            // randomly select from either the null fee-rate transactions or those with fee-rate estimates.
+                            let opt_tx = if start_with_no_estimate {
                                 null_iterator
                                     .next()
                                     .map_err(|err| Error::SqliteError(err))?
-                            } {
-                                Some(row) => (
-                                    MemPoolTxInfoPartial::from_row(row)?,
-                                    !start_with_no_estimate,
-                                ),
+                            } else {
+                                fee_iterator.next().map_err(|err| Error::SqliteError(err))?
+                            };
+                            match opt_tx {
+                                Some(row) => (MemPoolTxInfoPartial::from_row(row)?, start_with_no_estimate),
+                                None => {
+                                    // If the selected iterator is empty, check the other
+                                    match if start_with_no_estimate {
+                                        fee_iterator.next().map_err(|err| Error::SqliteError(err))?
+                                    } else {
+                                        null_iterator
+                                            .next()
+                                            .map_err(|err| Error::SqliteError(err))?
+                                    } {
+                                        Some(row) => (
+                                            MemPoolTxInfoPartial::from_row(row)?,
+                                            !start_with_no_estimate,
+                                        ),
+                                        None => {
+                                            debug!("No more transactions to consider in mempool");
+                                            break MempoolIterationStopReason::NoMoreCandidates;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                        MemPoolWalkStrategy::NextNonceWithHighestFeeRate => {
+                            match query_stmt_nonce_rank
+                                .query(NO_PARAMS)
+                                .map_err(|err| Error::SqliteError(err))?
```

Comment:
This, and several other `map_err()` calls in this function, can be simplified to:

```suggestion
                                .map_err(Error::SqliteError)?
```

or

```suggestion
                                .map_err(Error::from)?
```

Reply:
Nice, thanks, done